h/api/resources.py

```diff
@@ -22,12 +24,69 @@ def add(self, name, obj):
         self[name] = obj
 
 
-class Annotations(object):
+class Collection(Resource):
     """
-    Annotations is a container resource that exposes annotations as its
-    children in the tree.
+    A collection of other resources.
     """
-    def __init__(self, request):
+
+    def __getitem__(self, key):
+        if key not in self and callable(self.factory):
+            self.add(key, self.factory(key))
+        return super(Collection, self).__getitem__(key)
+
+    def factory(self, key):
+        raise NotImplementedError
+
+
+class Annotation(Resource):
+    """
+    A Resource representing an annotation.
+    """
+
+    def __acl__(self):
+        acl = []
+
+        model = self.model
+
+        # Convert annotator-store roles to pyramid principals
+        for action, roles in model.get('permissions', {}).items():
+            for role in roles:
+                if role.startswith('system.'):
+                    raise ValueError('{} is a reserved role.'.format(role))
+                elif role.startswith('group:'):
+                    if role == 'group:__world__':
+                        principal = Everyone
+                    elif role == 'group:__authenticated__':
+                        principal = Authenticated
+                    elif role == 'group:__consumer__':
+                        raise NotImplementedError("API consumer groups")
+                    else:
+                        principal = role
+                else:
+                    principal = role
+
+                # Append the converted rule tuple to the ACL
+                rule = (Allow, principal, action)
+                acl.append(rule)
+
+        return acl
+
+    @reify
+    def model(self):
+        if 'id' in self:
+            instance = models.Annotation.fetch(self['id'])
+            return instance or models.Annotation()
```

Comment:
Yeah, this is pretty odd. It means that the only reason I get a 404 at `/a/gibberish` is because a blank annotation doesn't give the `read` permission.